File: backend/management/src/main/java/dev/tpcoder/empfriendly/management/point/PointController.java
package dev.tpcoder.empfriendly.management.point;

import javax.validation.Valid;
import javax.validation.constraints.NotBlank;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.ResponseStatus;
import reactor.core.publisher.Mono;

@Controller
@ResponseBody
@RequiredArgsConstructor
@RequestMapping("/points")
public class PointController {

  private final PointService pointService;

  @PostMapping("/topup")
  @ResponseStatus(HttpStatus.NO_CONTENT)
  public Mono<Void> topup(@RequestBody @Valid PointChangeRequest body) {
    return pointService.topup(body).then();
  }

  @PostMapping("/deduct")
  @ResponseStatus(HttpStatus.NO_CONTENT)
  public Mono<Void> deduct(@RequestBody @Valid PointChangeRequest body) {
    return pointService.deduct(body).then();
  }

  @GetMapping("/{refId}")
  public Mono<Point> getPointByRefId(@PathVariable @NotBlank String refId) {
    return pointService.getPointByRefId(refId);
  }
}

File: backend/general/src/main/kotlin/dev/tpcoder/empfriendly/general/point/PointController.kt
package dev.tpcoder.empfriendly.general.point

import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestHeader
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.ResponseBody
import reactor.core.publisher.Mono

@Controller
@ResponseBody
@RequestMapping("/points")
class PointController(private val pointService: PointService) {

    @GetMapping("/me")
    fun getPoint(@RequestHeader("user-id") userId: String): Mono<Point> {
        return pointService.getPointByRefId(userId)
    }
}